two-entity-poc/src/controllers/blog.ts

```diff
@@ -0,0 +1,74 @@
+import { NextFunction, Request, Response } from "express";
+import { IBlog } from "../interfaces/blog-interface";
+import BlogService from "../services/blog";
+import { IUser } from "../interfaces/user-interface";
+import { ERROR_MESSAGES, SUCCESS_MESSAGES } from "../constants/controller-messages";
+import { getErrorObject } from "../util/generate-error";
+
+const blogController = {
+
+    addBlog: async (req: Request<{}, {}, { title: string; content: string; imageUrl: string; user: IUser }>, res: Response, next: NextFunction) => {
+        try {
+            const { title, imageUrl, content, user } = req.body;
+            if (!user) {
+                return res.status(401).json({ error: ERROR_MESSAGES.USER_NOT_AUTHENTICATED });
+            }
+            const blog: IBlog = { title, imageUrl, content };
+            const createdBlog = await BlogService.addBlog(user, blog);
+            res.status(201).json({ blog: createdBlog });
+        } catch (error) {
+            console.error(error);
+            next(error);
+        }
+    },
+
+    getBlogDataById: async (req: Request<{ blogId: string }, {}, { user: IUser }>, res: Response, next: NextFunction) => {
```

Comment:
In your next project.
Try using Zod for request body validation; or if you can find time, you can implement for certain route in this project also

Reply:
Sure will explore validation using zod in next assignment.